### PR TITLE
Fix bugs in kurobako/solver/optuna.py

### PR DESCRIPTION
**Pull request by contramundum53**
Several bugs were present in `kurobako/solver/optuna.py` that resulted in errors when used with nontrivial pruners.

* `trial.last_step` is undefined when `trial` is not a `FrozenTrial`.
* `self._study._log_completed_trial` requries a `FrozenTrial`, but a `Trial` is passed
* Logs for pruning are invisible

These bugs are fixed in this PR.

### CODE REVIEW

File: kurobako/solver/optuna.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from kurobako import problem
 from kurobako import solver
 
-_optuna_logger = optuna.logging.get_logger(__name__)
+_optuna_logger = optuna.logging.get_logger("optuna.study.study")
```
Comment (himkt) [inline kurobako/solver/optuna.py]: Is this change really needed?

File: kurobako/solver/optuna.py [hunk 2]
```diff
@@ -180,8 +181,8 @@ def tell(self, evaluated_trial: solver.EvaluatedTrial):
 
         assert current_step <= self._problem.last_step
         if self._problem.last_step == current_step:
-            self._study.tell(trial, values=values)
-            self._study._log_completed_trial(trial, values)
+            frozen_trial = self._study.tell(trial, values=values)
+            self._study._log_completed_trial(frozen_trial)
```
Comment (sile) [inline kurobako/solver/optuna.py]: The problem here was addressed in https://github.com/optuna/kurobako-py/pull/19, and the PR has been merged. So, please sync with the master branch 🙏

File: kurobako/solver/optuna.py [hunk 3]
```diff
@@ -12,9 +12,12 @@
 
 from kurobako import problem
 from kurobako import solver
+import logging
 
 _optuna_logger = optuna.logging.get_logger(__name__)
-
+_optuna_logger.setLevel(optuna.logging.INFO)
+console = logging.StreamHandler()
+_optuna_logger.addHandler(console)
```
Comment (HideakiImamura) [inline kurobako/solver/optuna.py]: Could you explain the necessity of this change?

File: kurobako/solver/optuna.py [hunk 4]
```diff
@@ -182,11 +186,19 @@ def tell(self, evaluated_trial: solver.EvaluatedTrial):
...
+                        trial.number, current_step, values[0], self._study.best_value
+                    )
+                )
             else:
-                self._study._log_completed_trial(trial, values)
+                _optuna_logger.info(
+                    "Successful trial#{}: step={}, value={}".format(
+                        trial.number, current_step, values
+                    )
+                )
```
Comment (HideakiImamura) [inline kurobako/solver/optuna.py]: Ditto.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I am wondering if `Study._log_completed_trial` could be used. Could you explain the change a little more?

File: kurobako/solver/optuna.py (hunk 3 above)
Comment (toshihikoyanase) [reply kurobako/solver/optuna.py]: In my understanding, handlers and log levels are usually set in the user code, not library code.
I don't think it is a good practice to show log messages.

**Comment by HideakiImamura** [comment]
I privately discuss above with @contramundum53 and @sile. As a result, we have decided to generate a log in our own format for the completed trials in this PR. The reason are
- to be in the same format as pruned trials, and
- note to use `_log_completed_trial`, which is a private method of optuna.

**Comment by contramundum53** [comment]
We reverted changes about `_optuna_logger` and made them a separate issue (#20).

**Review by sile — APPROVED** [review]
Thanks, LGTM!
